typescript/sdk/src/router/HyperlaneRouterChecker.ts

```diff
@@ -13,13 +18,24 @@ import {
   ConnectionClientViolationType,
   OwnableConfig,
   RouterConfig,
+  RouterViolation,
+  RouterViolationType,
 } from './types';
 
 export class HyperlaneRouterChecker<
   Factories extends HyperlaneFactories,
   App extends RouterApp<Factories>,
   Config extends RouterConfig,
 > extends HyperlaneAppChecker<App, Config> {
+  constructor(
+    multiProvider: MultiProvider,
+    app: App,
+    configMap: ChainMap<Config>,
+    readonly ismFactory?: HyperlaneIsmFactory,
```

Comment:
Why is only the `ismFactory`readonly?

Reply:
mostly wanted the ability to do `this.ismFactory`, and I don't see it as ever needing to be reassigned. Would you prefer to change?